src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -666,6 +666,53 @@ object TezosDatabaseOperations extends LazyLogging {
     }
   }
 
+  /** Updates an account by id, marking it with a flag.
+    * @param id the account identifier on database (corresponds to the pkh)
+    * @return the action to run against the db, which will tell how many rows where touched
+    */
+  def activateAccount(id: String): DBIO[Int] =
+    Tables.Accounts
+      .findBy(_.accountId)
+      .extract(id)
+      .map(_.isActivated)
+      .update(true)
+
+  /** Updates an account history by id and asof, marking it with a flag.
+    * @param id the account identifier on database (corresponds to the pkh)
+    * @param asof which flag to set "on"
+    * @return the action to run against the db, which will tell how many rows where touched
+    */
+  def activateAccountHistory(ids: List[String], asof: java.sql.Timestamp): DBIO[Int] =
+    Tables.AccountsHistory
+      .filter(acc => acc.asof >= asof && acc.accountId.inSet(ids))
+      .map(_.isActivated)
+      .update(true)
+
+  /** Finds activated accounts - useful when updating accounts history
+    * @return sequence of activated account ids
+    */
+  def findActivatedAccounts: DBIO[Seq[String]] =
+    Tables.Accounts
+      .filter(_.isActivated)
+      .map(_.accountId)
+      .result
+
+  /** Load all operations referenced from a block level and higher, that are of a specific kind.
+    * @param ofKind a set of kinds to filter operations, if empty there will be no result
+    * @param fromLevel the lowest block-level to start from, zero by default
+    * @return the matching operations, sorted by ascending block-level
+    */
+  def fetchRecentOperationsByKind(
+      ofKind: Set[String],
```

Comment:
Do we have typed enumerations of the existing operation kinds somewhere, to use instead of treacherous strings? 😁

Reply:
nope, I did not find that, we have a trait which extends case classes